lib/netext/httpext/request.go

```diff
@@ -181,13 +181,18 @@ func MakeRequest(ctx context.Context, state *lib.State, preq *ParsedHTTPRequest)
 		}
 	}
 
-	tracerTransport := newTransport(ctx, state, preq.Tags, preq.ResponseCallback)
+	tracerTransport := newTransport(ctx, state, &preq.TagsAndMeta, preq.ResponseCallback)
 	var transport http.RoundTripper = tracerTransport
 
 	if state.Options.HTTPDebug.String != "" {
 		// Combine tags with common log fields
 		combinedLogFields := map[string]interface{}{"source": "http-debug", "vu": state.VUID, "iter": state.Iteration}
-		for k, v := range preq.Tags.Map() {
+		for k, v := range preq.TagsAndMeta.Tags.Map() {
+			if _, present := combinedLogFields[k]; !present {
+				combinedLogFields[k] = v
+			}
+		}
+		for k, v := range preq.TagsAndMeta.Metadata {
```

Comment:
Just in case, we should probably reverse the order here - first add the Metadata entries and then the Tags. That way, if there is a duplicate key, the tag value will be shown (which is probably more important :thinking:

Reply:
Should we split them into two sub-json-objects? I think could be relevant to see both, especially in debugging phases.

```json
{
  // ...
  "iter": 10,
  "tags": {"key1":"val1",...},
  "meta": {"meta1":"val1"}
}
```

Reply:
I don't think this is possible, and even if it is, it won't look nice with logrus. Since it's such a minor concern and only matters for `--http-debug`, I think we'd be fine with the current solution, just flipping the tags.